index.js

```diff
@@ -392,17 +379,12 @@ let Fixtures = {
      * @return {Promise} The ChakramResponse for the delete request to the server.
      */
     cleanup() {
-        if (_.isUndefined(this._headers['OAuth-Token'])) {
-            if (++this._sessionAttempt > this._maxSessionAttempts) {
-                throw new Error('Max number of login attempts exceeded!');
-            }
-
-            return this._adminLogin().then(() => this.cleanup());
+        if (this._needsLogin()) {
```

Comment:
> Also, note that for whatever reason we don't do this same check on Fixtures.link. I don't remember whether that was a deliberate decision or not.

Whether it was a deliberate decision or not, it looks like we have to. See below.

```
Request  146 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/bulk
Response 146 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/bulk 200
Request  147 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/bulk
Response 147 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/bulk 200
Request  148 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9bdb6aa-8453-11e7-b509-a0999b020fc9/link
Request  149 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9ff3ed6-8453-11e7-ae93-a0999b020fc9/link
Response 149 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9ff3ed6-8453-11e7-ae93-a0999b020fc9/link 401
Request  150 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token
Response 148 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9bdb6aa-8453-11e7-b509-a0999b020fc9/link 401
Request  151 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token
Response 150 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token 400
Request  152 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9ff3ed6-8453-11e7-ae93-a0999b020fc9/link
Response 151 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token 400
Request  153 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9bdb6aa-8453-11e7-b509-a0999b020fc9/link
Response 152 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9ff3ed6-8453-11e7-ae93-a0999b020fc9/link 401
Response 153 Fixtures: POST /TY-1998/ent/sugarcrm/rest/v10/Contacts/d9bdb6aa-8453-11e7-b509-a0999b020fc9/link 401
Request  154 Agent-John1503088131292: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token
Response 154 Agent-John1503088131292: POST /TY-1998/ent/sugarcrm/rest/v10/oauth2/token 200
Request  155 Agent-John1503088131292: GET /TY-1998/ent/sugarcrm/rest/v10/Contacts
Response 155 Agent-John1503088131292: GET /TY-1998/ent/sugarcrm/rest/v10/Contacts 200
```